### PR TITLE
Add Workspace name validation (#834)

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by TristanCacqueray** [comment]
Thanks, that looks great to me, I meant to do that multiple times. Would you mind adding an entry to the CHANGELOG please.

File: src/Monocle/Effects.hs [hunk 1]
```diff
@@ -99,6 +99,7 @@ import Effectful.Reader.Static qualified as E
 import Effectful.Retry as Retry
 import Monocle.Client (MonocleClient)
 import Monocle.Client.Api (crawlerAddDoc, crawlerCommit, crawlerCommitInfo)
+import Monocle.Config (IndexName, mkIndexName)
```
Comment (morucci) [inline src/Monocle/Effects.hs:102]: CI failure: hint suggests to group Monocle.Config imports:

```
  import Monocle.Config qualified                                                                                                                                                                                                                                   
  import Monocle.Config ( ConfigStatus, IndexName, mkIndexName )  
```

File: src/Monocle/Config.hs [hunk 2]
```diff
@@ -513,4 +480,15 @@ getIdentByAliasFromIdents alias idents' = case find isMatched idents' of
   isMatched :: Ident -> Bool
   isMatched Ident {..} = alias `elem` aliases
```
Comment (morucci) [inline src/Monocle/Config.hs:482]: Awesome !
Nit; Adding doctests would be a great addition too.

**Review by morucci — COMMENTED** [review]
Thanks for the contribution that looks great ! I did a quick manual test trying to load config with wrong index names and the errors are well detected.

Would you like to fix the lint issue and add the changelog entry as suggested by @TristanCacqueray  ?

**Review by TristanCacqueray — APPROVED** [review]

**Review by morucci — APPROVED** [review]

**Pull request merged**